org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -607,14 +608,19 @@ import org.eclipse.xtext.findReferences.IReferenceFinder.IResourceAccess
 		throw new UnsupportedOperationException("TODO: auto-generated method stub")
 	}
 
-	override rename(RenameParams params) {
+	override rename(RenameParams renameParams) {
 		return requestManager.runRead[ cancelIndicator |
-			val uri = params.textDocument.uri.toUri
+			val uri = renameParams.textDocument.uri.toUri
 			val resourceServiceProvider = uri.resourceServiceProvider
 			val renameService = resourceServiceProvider?.get(IRenameService)
 			if (renameService === null)
 				return new WorkspaceEdit
-			renameService.rename(workspaceManager, params, cancelIndicator)
+			if (renameService instanceof IRenameServiceExtension) {
+				val options = new IRenameServiceExtension.Options(params?.capabilities?.workspace?.workspaceEdit?.documentChanges === Boolean.TRUE)
+				renameService.rename(workspaceManager, renameParams, options, cancelIndicator)
```

Comment:
i still wonder if we can get rid of workspace manager here. e.g. codeaction does not have access to it too.
i also wonder if and how we should maintain api here since lsp/lsp4j changes theirs for each release

Reply:
@szarnekow @kthoms what do you think?

Reply:
Iff there was a customized IRenameService, it would no longer be called?

Reply:
@cdietrich 

@szarnekow why? The LS still injects the old interface and only calls the new method if it's present. Did I miss something?

Reply:
My feeling is that you missed that the old impl implements the original interface but no framework code will call it anymore.